### ISSUE TITLE
DataLoader, reconstruct.py 코드 변경 문의

### ISSUE THREAD

**Issue by hwater27** [LISTatSNU/FastMRI_challenge#355]
안녕하세요? 다음과 같이 두 가지 문의드립니다.

**1. DataLoader가 생성될 때 내부에서 classifier가 돌아가도록 코드를 만들어도 될까요?**
각 data의 분류 결과를 dataloader에 미리 저장해 두고 그때그때 빠르고 편하게 꺼내오려고 합니다. 사실 dataloader 자체는 변경이 가능한 대상이기도 하고 classifier를 통한 MoE도 허용되어서 안될 것도 없다고 생각했지만, 생각해보니 dataloader 생성시에는 아직 kspace와 image에 mask가 씌워지지 않은 상태더라구요. 그래서 저희가 자체적으로 조정하지 않는 이상 classifier에 원본 이미지가 들어가게 되는 것 같은데 괜찮은지 궁금합니다. 아니면 원본 이미지에 access가 가능한 상황에서 코드를 일부러 조정해서 dataloader 내부에서 mask를 미리 씌우고 classifier에 통과시킨 후 dataloader를 생성하는 방법이면 명분상 안될 건 없다고 생각하는데, 약간 형식적인 눈가리고 아웅 느낌이라서 어떻게 하면 될지 여쭤봅니다.

**2. 처음 받은 reconstruct.py와 leaderboard_eval.py 코드를 이용하여 점수를 내라고 하셨는데, reconstruct.py에 args 추가 및 약간의 코드 변경을 해도 될까요?**
저희가 모델을 생성할 때 들어갈 인수가 조금 필요하기도 하고, 현재 코드는 val loss 기준 best model만을 이용하여 reconstruct하도록 되어 있어 이 정도는 변경해도 괜찮을지 여쭤봅니다.

**Comment by B0neh3ad** [comment]
@hwater27 안녕하세요,

1. **inference 과정에서 ground truth image를 사용**하는 것은 보편적인 ML/DL 대회에서 **허용되지 않으며**, 본 대회도 마찬가지입니다. Data Explanation ppt를 보시면 각 image data는 아래의 세 data를 포함하고 있음을 아실 수 있습니다.
    >  Aliased image, GRAPPA reconstructed image, Aliasing free image (ground truth)
    
    이 중 **aliased image**나 **GRAPPA reconstructed image**를 classifier의 input으로 활용하시기 바랍니다
    kspace의 경우, training data에는 fully-sampled data가 주어지나 leaderboard data를 포함한 test data에는 undersampled data만 주어진다는 점 참고해주시기 바랍니다.

2. `reconstruct.py`는 이미지 shape만 적절하다면 **자유롭게 변경하셔도 됩니다.**
    아래는 저희가 Final Submission 설명을 할 때 사용한 슬라이드인데, 아마 이 부분을 보시고 그렇게 생각하신 듯한데요. 영상에서는 `leaderboard_eval.py`에 대해서만 임의변경 금지를 말씀드렸으며(해당 코드가 SSIM을 계산하므로) `reconstruct.py`에 대해서는 별도의 언급을 하지 않았습니다. 따라서 말씀하신 정도의 변경, 혹은 그 이상도 얼마든지 가능합니다.
<img width="300" alt="Image" src="https://github.com/user-attachments/assets/a91e5595-b45c-46db-806b-3d4508f11e6e" />

감사합니다.
